### DISCUSSION TITLE
Controller Board

### DISCUSSION THREAD

**Discussion by BobbyPr** [missionpinball/missionpinball#103] [Q&A]
How do you determine the best controller to use? There are so many! FAST, OPP, P-ROC, LISY etc... 
My goal is only to recreate some simple classic games like Eight Ball, Eight Ball Deluxe, Joker Poker, possibly some multi-level games like Mars Attacks

**Comment by toomanybrians** [comment]
Disclaimer: I am part of FAST Pinball.

Good question and probably something we should add to the docs!

The first thing I would ask is whether you intend start with an existing machine and reuse as much of the electronics and wiring as you can, or are you building a machine with all new electronics? (Using “all new electronics” still could be with an old machine, but the question is, are you going to rip out all the guts and rewire from scratch? Or are you going to just drop in a modern pinball controller and use the existing legacy driver boards, wiring, etc.)

If you just want to drop in a new controller and reuse all of the existing wiring, then that will dictate your choice. e.g. an existing WPC or System 11 machine would use a FAST Retro Controller, older Gottlieb machines would be LISY, Stern SAM or Whitestar would be P-ROC, etc.

If you’re going to do new electronics and wiring (whether an existing old machine or scratch-built homebrew), then of course you have much more choice.

I would say that how “complex” your machines are doesn’t really matter. Like, if you said, “I’m just trying to do a simple 1980s single level playfield” or “I want to create a 2020s modern machine that’s off the charts!”, those are both great options, but I don’t think that would affect which hardware you use. All pinball control systems today basically just manage the pinball elements, and your host PC running MPF controls the LCD, graphics, sounds, etc. And frankly whether your machine has 16 drivers or 32 drivers, and 20 switches or 100 switches, probably doesn’t matter too much. (Obviously check to see what the max number of drivers and switches the various platforms can support, but other things like servos and LEDs and all that are easy to add-in to anything.)

Some people try to pick a platform based on price, though honestly building a pinball machine is so expensive and takes so much time that saving a few hundred dollars in up front hardware pricing doesn’t end up being too significant once you look at the overall cost and time of your total project.

Also when comparing pricing, you need to consider what’s included in the price, and what external components you would need or not need with a certain platform. Using FAST Pinball as an example (again since I’m part of them), our stuff is on the more expensive side when you compare the price of controllers, but also the FAST stuff has a lot of protections built in and it’s much more rare to fry a whole board at once. So if you compare a $159 driver board versus an $80 one, obviously one is 2x the cost, but if the cheap one gets blown up because you bumped something with a screwdriver causing you to have to buy another one, now the “cheaper” board is the same cost plus the hassle of needing to order a replacement, etc.

Also consider how granular the pricing options are. e.g. can you add 4 new drivers at a time, or do you have to add them in blocks of 16?

The other thing about pricing is you need to look at what’s included in that cost. Using another FAST example, our current Neuron platform does not use a 5-volt power system (rather it’s all 12V and then the individual boards generate whatever voltages they need from that.) That means no 12V PSU (saving money), no additional wiring for the 12V power, plus you have more voltage at the board level so less need to run additional power taps to top up the power, etc. All those costs add up. Same for the “other” things that might be included. Does the platform or boards include their own wiring distribution, or are you buying terminal strips and wire connectors and designing all that yourself too, or is that included in what you get from your controller vendor?

Another factor in selecting a platform would be what your goals for your project are. The FAST Platform is designed with commercially released machines in mind, which means if you follow our guidance, a machine you build on FAST would pass UL listing, FCC and CE emissions requirements, etc. I don’t think I’ve seen that as something that other platforms talk about, though again if you’re just building a machine for your own basement then maybe you don’t care?

I would also look at the various manufacturer websites and see how much technical documentation and guidance they have. If you choose a platform, are they going to show you how to wire everything up, or are you on your own for “figuring it out.” (And if you have to figure it out on your own, will that increase the chance you accidentally blow something up and have to buy a replacement which will eat into the savings of buying the cheap stuff from the start?)

Also consider what support & community options are available. Again since I’m part of FAST I can only comment on the FAST community, but we have a Slack group for people building projects on FAST, and there are something like 200 people in that group and it’s VERY active, with channels for hardware building, MPF, wiring, community, etc, etc. I think other platforms have things like that too but don’t know the details.

For MPF in general these days, it seems like most new projects are either FAST or CobraPin. P-ROC / P3-ROC was probably the most popular 5-10 years ago, but they seem more focused on the P3 video pinball platform now versus homebrew. All the other platforms (LISY, APC…) seem to only have a small handful of products?

Oh, I would also look at the hardware vendor support for MPF as well. For example, even though I’m FAST, I spend a lot of contributing to MPF, so you know the MPF/FAST interface will always be solid. The same is true for the CobraPin/OPP stuff as they contribute to MPF as well. But I don’t think I’ve seen as much or anything from the other vendors?

Bottom line though is that pinball machines have been built with all hardware, and the most important thing is that you start building some pinball!! :)

Brian

**Reply by BobbyPr** [reply]
Thank you for that highly detailed response! wasn't really expecting anything, but that response, and so fast, is great. 
To respond to your question, I'll be building from scratch. And FAST can thank you because I'll go with them now, simply on your response!
Ready the MPF docs is really complicated. Links that go to links that go to videos.... seems like a giant rabbit hole, and my friend and I are not quit convinced we can do this yet. We just finished a virtual pinball cabinet that works great and has all the bells and whistles, but now reading about MPF, it seems like this is the next level. Our long term end game is to have one cabinet that we can interchange playfields / games. We are both in our 60's and like the older games.
So far, 've been following along with the tutorials (I'm only on the fourth one) but I want to see if there is a way that we can slowly commit to this in terms of cost. ie what board(s) can we buy / start with just to see if we are on the right track with all this in terms of this actually working. We would just like to prove to ourselves that we can do something as simple as write some program that activates a switch that scores points. I think if we do that, we would gain confidence and therefore commitment.  
Any specific recommendations in terms of specific initial hardware purchases and a path to future ones?
We think we can build the actual coils ourselves.
Again, thanks for the fast response!

**Reply by BobbyPr** [reply]
sorry just reading more now. I see that FAST has a starter kit that looks like what I would need, and there is an unpacking video that is great (I think that it's you in it?)  but holy cow by the time I put shipping and everything it's $644 USD. I'm in Canada so that's close to $1G.
Is this what I'm going to eventually need? Will have to discuss with my friend but that is a lot more than what I thought :(

**Comment by toomanybrians** [comment]
MPF can definitely be overwhelming. Luckily hundreds of people have gone down this path, and many of them didn't know too much when they began. A few weeks ago at the Northwest Pinball show we held a little pinball maker event called "PinDevCon." Part of that was a series of 10-minute interviews with homebrew makers. They'll all on youtube, but I would encourage you to watch some of these. Aaron asks most of them about how they got over their "hard parts" of the build, and how they had the confidence to begin, etc. Many people interviewed said similar things like "I wasn't even sure I had the skills to do this project!"

Check out those homebrew chat videos on the PinDevCon page. They're the videos with the colored backgrounds at the bottom of this page: https://fastpinball.com/pindevcon/

Also btw there's a pinball maker named Brian Cox (he was one of the guys interviewed in the videos above) who did a project called "Brian's Any Machine" which is similar to what you're talking about. He built one cabinet and then swaps out playfields. We actually did a little interview with him at the show two years ago here: https://www.youtube.com/watch?v=G6ndvEqj4x0 Also check out his website: http://space-eight.com/pinball.html

Oh, speaking of videos about making pinball machines, if you haven't seen it yet, The Pinball Room channel on YouTube is amazing. Steve has something like 75 videos in his "How to Make a Pinball Machine" series which walk you through all aspects of building a machine from scratch, including lots of videos on how to do things in MPF. That series is here:

https://www.youtube.com/watch?v=T8uP0TZC9Ak&list=PL4ICKQRfLA2aB8D8Jqw-e70Bfc8erUNp7

I do think it would be good to add some MPF documentation to walk through the high-level process of actually writing a game in MPF. I agree there are lots of great individual docs, but there needs to be a high level roadmap or something.

One cool thing about MPF is you don't need a pinball machine to start. Just use MPF plus the MPF-Monitor (you don't even need to worry about the media control) and use MPF's smart virtual platform and you can get MPF running enough for you to be able to start a game and score some points.. that should let you know if it's something you'd like to commit to for real.

Finally, for cost of boards, yes, things get expensive when you add everything up. But again, make sure what you're including in your "holy cow" reaction is the same between different hardware platforms. Does that $1k hardware cost include power supplies? (If not, which power supplies do you need? Any how many?) Does it include power distribution? Does it include connectors? Does the hardware require more expensive cables? Do you need additional boards for what you want to do? Can the cheaper hardware mix and match voltages, LED types, or switch types? (Or does that not matter for you?) Are you using magnets, and if so do you need additional hardware for them? Does the hardware cost include PC or machine power control, or are you buying another board for that? Do you need to control an LCD? Or segment displays? How are you handling sound? Does the cheaper hardware have instructions for wiring or will you blow up a few boards which you'll have to replace? Where are the switch and driver boards located, and how much wire do you need for one board set versus another? (Wire is expensive, and where your boards are placed can affect the cost of wiring by more than $100 for a single machine!) Do you have to make all your own cables or can you use off the shelf ones? Do you want to deal with switch matrix wiring or only go direct? Will you have high voltage in the cabinet? What extra costs will be needed to support that? Do you need to eliminate EMI and interference? (Or do you even care?) Where's your knocker going (or do you even care)? Will there be a shaker? How much lighting? Do you care about RGBW (with soft "real" white light) or are RGB with fake white light ok? Do you want to mix and match lighting types? Is open hardware important to you, or is proprietary commercial ok? Are you comfortable working on SMT components, or do you want to stick with sockets and through-hole? Etc, etc....

All of this affects the "total cost" of the machine and is affected by which hardware boards you choose. But since you're just beginning, at this point, you don't know what you don't know. So I would just go with what your gut says for now, and by the time you start your second machine you'll know what's most important for you! But also the $1k in hardware is just a drop in the bucket of what you'll spend on this project overall. 🤣
